### PR TITLE
tests: get the linux-image-extra available for the current kernel

### PR DESCRIPTION
**Pull request by sergiocazzolato**
This is to fix the missmatch ith kernels without a linux-image-extra
with the same version.

This is currently happening on ubuntu-16.04-64. See the error:
https://paste.ubuntu.com/p/34hnHyHYp6/

### CODE REVIEW

File: tests/lib/pkgdb.sh [hunk 1]
```diff
@@ -562,11 +558,22 @@ pkg_dependencies_ubuntu_classic(){
     esac
 }
 
+pkg_linux_image_extra(){
+    available_pkgs=$(apt-cache policy linux-image-extra-*-generic | grep -E 'linux-image-extra-.*:' | tr -d :)
```
[the remark this answers is not in the document]
Comment (sergiocazzolato) [reply tests/lib/pkgdb.sh]: @bboozzoo it is not working at all. 
The weird part is that I execute the current function in ubuntu-14.04-64 and works properly

both functions: https://paste.ubuntu.com/p/z6vdqXQXdt/

File: tests/lib/pkgdb.sh [hunk 2]
```diff
@@ -562,11 +558,22 @@ pkg_dependencies_ubuntu_classic(){
     esac
 }
 
+pkg_linux_image_extra(){
+    if apt-cache policy "linux-image-extra-$(uname -r)" >/dev/null 2>&1; then
```
Comment (mvo5) [inline tests/lib/pkgdb.sh]: Thanks, this is much cleaner than the previous version. You could use `apt list "linux-image-extra-$(uname -r)"` here but this is fine.

**Comment by sergiocazzolato** [comment]
Sadly I had to revert the last change proposed by @bboozzoo because it is not working
see logs: https://api.travis-ci.org/v3/job/416868239/log.txt

The first "apt-cache policy ...." always return 1 and it produces that the linuximage-extra-4.13.0-1019-gcp package is being returned, but this package is not in the repository.

File: tests/lib/pkgdb.sh [hunk 3]
```diff
@@ -559,14 +559,14 @@ pkg_dependencies_ubuntu_classic(){
 }
 
 pkg_linux_image_extra(){
-    available_pkgs=$(apt-cache policy linux-image-extra-.*-generic | grep -E 'linux-image-extra-.*:' | tr -d :)
-    if echo "$available_pkgs" | grep -q "linux-image-extra-$(uname -r)"; then
-        echo "linux-image-extra-$(uname -r)"
-        return
+    if apt-cache policy "linux-image-extra-$(uname -r)" >/dev/null 2>&1; then
+        echo "linuximage-extra-$(uname -r)"
```
Comment (bboozzoo) [inline tests/lib/pkgdb.sh]: @sergiocazzolato there's a typo here in `echo "linuximage-extra-$(uname -r)"`, missing space, should be `echo "linux-image-extra-$(uname -r)"`

Comment (sergiocazzolato) [reply tests/lib/pkgdb.sh]: @bboozzoo, I'll test that right now

**Pull request merged**